### DISCUSSION TITLE
Dealing with networks with over 100/150 devices

### DISCUSSION THREAD

**Discussion by andyshilton** [Koenkk/zigbee2mqtt#5455] [Devices]
I’m currently running two hue hubs and a CC1352P-2 for zigbeetomqtt. I have the two hue hubs because the whole house is pretty much running hue, but I’ve added tons of aqara sensors and a few innr plugs and innr bulbs using zigbeetomqtt - all plumbed into Home Assistant.

I bought the CC1352P-2 because it supports “100’s of devices” but, I sat down to move my hue bulbs across to z2m and when I hit 100 devices, it started dropping (well, losing comms with) devices I had added before. So I think the max children it can support is 100. So I took a load of devices back out to try joining them via routers…

I tried the new “permit join” via a specific device, trying hue bulbs as the connection device (router) and innr plugs. I tried connecting a hue bulb and an aqara battery powered switch both via both. None of the four combinations worked so I’ve not been able to add more than 100 devices to my network.

I’m really keen to get this to work because the z2m network is SO stable, whereas the hue hubs connections drop all the time.

My question is really for advice/help - has anyone managed to extend their network passed the max direct children? I’d be extremely grateful for suggestions and help.

**Comment by jarrah31** [comment]
I too would like to know more about this please.

Given that [this page](https://github.com/Koenkk/Z-Stack-firmware/tree/master/coordinator) says you can have 200 devices, how do we achieve this using a CC1352-P or zsh's CC2652R? It says Source Routing is the answer, but how do we do that within Z2M? 

At the moment I have two zigbee2MQTT networks running from two Raspberry Pis, with one containing 49 devices and the other 42.

I also have a few SA-003-Zigbee smart plugs that act as routers - does this mean I can go beyond the 50 direct children limit?

Is it possible to have these limits and ways to expand them explained within the https://www.zigbee2mqtt.io documentation please? 

Thanks in advance.

**Reply by Koenkk** [reply]
> Given that this page says you can have 200 devices, how do we achieve this using a CC1352-P or zsh's CC2652R? It says Source Routing is the answer, but how do we do that within Z2M?

Source routing only applies to the CC2531 (due to memory constraints), the CC2652/CC1352 all have source routing enabled by default.

> I also have a few SA-003-Zigbee smart plugs that act as routers - does this mean I can go beyond the 50 direct children limit?

Yes

**Reply by jarrah31** [reply]
Thank you @koenkk for clarifying that - much appreciated! 

Just a quick side question, which device is better to act as my main coordinator for 100+ devices - the CC2652 (zzh) or LAUNCHXL-CC1352P-2 module?

**Reply by Koenkk** [reply]
No difference, hardware is almost identical

**Comment by Koenkk** [comment]
@andyshilton are you still interested in this? If you could take some sniff when trying to pair and it fails I can look into it: https://www.zigbee2mqtt.io/how_tos/how_to_sniff_zigbee_traffic.html

**Reply by andyshilton** [reply]
@Koenkk  Yes very much so. It would be amazing to just run one network with all my devices on. I'm just trying to upgrade my repo then I'll give it a go when I get some time in the next week or so.

I can tell you that _nothing_ happens when I try and add devices via others. (not that I can see in the z2m logs in HA anyway).

**Reply by Koenkk** [reply]
@andyshilton the sniff should reveal more information on why you don't see anything in the z2m log.

**Reply by andyshilton** [reply]
Just to keep this alive as I've not forgotten about it, I've managed to add a lot more devices. However with the one co-ordinator I'm finding that battery devices become quite unreliable until removed and re-added. I have now bought a couple of CC1352P-2 boards to flash as routers.

My current issue is that I cannot find instructions to add CC1352P-2 routers to my network. I've put it in join mode and power cycled, pressed buttons, etc, but it never joins the network.

**Comment by thargy** [comment]
Sorry for hijacking this discussion, but it feels relevant as I've currently got 114 devices added, and have about 50+ more to go. Unfortunately, I'm now experiencing issues adding more.  
<details>

<summary>Expand for full system stats</summary>
<details>
<summary>Properties</summary>

| Property  | Value  | 
| - | - | 
| Hardware | Beelink EQ12 Pro Intel 12th N305 |
| RAM | 16G (DDR5) |
| Zigbee2MQTT version | 1.32.0 |
| Coordinator | Sonoff Zigbee 3.0 USB Dongle Plus |
| Coordinator Chip | CC2652P |
| Coordinator type | zStack3x0 |
| Coordinator revision | 20221226 |
| Frontend version | 0.6.129 |
|Total Devices| 114|
</details>
<details>
<summary>By device type</summary>

| Device Type | Count |
| - | - |
|Router| 111|
|End devices| 3|
</details>
<details>
<summary>By power source</summary>

| Power Source | Count |
| - | - |
|Mains (single phase)| 114|
</details>
<details>
<summary>By vendor</summary>

| Vendor | Count |
| - | - |
|Philips| 77|
|_TZE204_ztc6ggyl| 15           |
|Sunricher| 9|
|LUMI| 3|
|_TZE200_fzo2pocs| 3|
|Signify Netherlands B.V.| 2|
|Candeo| 2|
|_TZB210_417ikxay| 1|
|_TZE200_aoclfnxz| 1|
|SONOFF| 1|
</details>
<details>
<summary>By model</summary>

| Model | Count |
| - | - |
|LCG002| 44|
|LTG002| 30|
|TS0601| 19|
|ZGRC-TEUR-004| 4|
|lumi.switch.l1aeu1| 3|
|LCE002| 3|
|HK-DIM-A| 2|
|ZGRC-TEUR-001| 2|
|ZGRC-TEUR-002| 2|
|LCA001| 1|
|LLC011| 1|
|ZGRC-TEUR-003| 1|
|TS0505B| 1|
|01MINIZB| 1|

</details>
</details>

> Just to keep this alive as I've not forgotten about it, I've managed to add a lot more devices. 

Any tips on how? As you can see from my stats, I'm mostly (77) using Hue bulbs (on the latest firmware), which are usually rock solid, but I'm now having difficulties adding more.  I avoid battery devices like the plague, so 111 of the 114 are powered routers.

> However with the one co-ordinator I'm finding that battery devices become quite unreliable until removed and re-added. I have now bought a couple of CC1352P-2 boards to flash as routers.

You mention you use Hue bulbs, aren't they already routers?  Is there any advantage in just adding a CC1352P-2 board configured as a router.  As I mentioned, most of my devices are routers already  (all but 3).

I'm aiming at around 200 devices, but hoping I can eventually exceed that.  @Koenkk  is there currently any way of going past 200 devices?  I appreciate the irony of asking that when I'm currently struggling around 114!

**Reply by jarrah31** [reply]
One suggestion is to have two zigbee networks, with each on a different channel that doesn't clash. I have two with each running around 75 devices, all without any issues so far. 
I've used this to help find the right zigbee/wifi channels that don't clash: https://www.metageek.com/training/resources/zigbee-wifi-coexistence/

**Reply by thargy** [reply]
> One suggestion is to have two zigbee networks, with each on a different channel that doesn't clash. I have two with each running around 75 devices, all without any issues so far.

Thanks.  I currently run Z2M as a Home Assistant addon, am I right in believing that you can only have on coordinator per Z2M instance, so this would necessitate somehow adding another Z2M instance (which I'm not sure is easy when you're running HAOS)?  I was hoping I could reach the alleged 200 devices without such hoops though.

> I've used this to help find the right zigbee/wifi channels that don't clash: [metageek.com/training/resources/zigbee-wifi-coexistence](https://www.metageek.com/training/resources/zigbee-wifi-coexistence/)

The Metageek Wi-Spy + Chanalyzer costs £500+ per year, which I can't really afford.  Plus, I made the mistake of using a Google Wifi Pro 6e mesh, which is likewise expensive junk that doesn't allow you to configure the wi-fi channels (amongst numerous other lacking features).

**Reply by jarrah31** [reply]
oops sorry, I didn't mean the hardware device, yeah I couldn't afford that either. :) I meant the wifi and ZigBee channel graphs on the page:

![](https://support.metageek.com/hc/article_attachments/115016951187/Wi-Fi_Channels.png)

![](https://support.metageek.com/hc/article_attachments/115016160948/ZigBee_Channels.png)

I run the two networks on two Raspberry Pi's, with each one sending MQTT messages to an MQTT broker (which can be hosted on HA I think).

**Reply by thargy** [reply]
> I run the two networks on two Raspberry Pi's, with each one sending MQTT messages to an MQTT broker (which can be hosted on HA I think).

Yeah, I was thinking if I can't make progress I could use the Pi that I previously ran HASS on.  I assume each network would be best on its own channel too.  It just seems a real shame to go that route when, theoretically, Zigbee networks should be able to support thousands of devices not just 200 max.  I read somewhere DeConz was supporting ~500, so it must be possible.

**Comment by seh-len** [comment]
Is it possible to activate source routing of the ConBee III from zigbee2mqtt?
